model/labels/labels.go

```diff
@@ -376,27 +381,111 @@ func FromStrings(ss ...string) Labels {
 // Compare compares the two label sets.
 // The result will be 0 if a==b, <0 if a < b, and >0 if a > b.
 func Compare(a, b Labels) int {
-	l := len(a)
-	if len(b) < l {
-		l = len(b)
+	l := len(a.lbls)
+	if len(b.lbls) < l {
+		l = len(b.lbls)
 	}
 
 	for i := 0; i < l; i++ {
-		if a[i].Name != b[i].Name {
-			if a[i].Name < b[i].Name {
+		if a.lbls[i].Name != b.lbls[i].Name {
+			if a.lbls[i].Name < b.lbls[i].Name {
 				return -1
 			}
 			return 1
 		}
-		if a[i].Value != b[i].Value {
-			if a[i].Value < b[i].Value {
+		if a.lbls[i].Value != b.lbls[i].Value {
+			if a.lbls[i].Value < b.lbls[i].Value {
 				return -1
 			}
 			return 1
 		}
 	}
 	// If all labels so far were in common, the set with fewer labels comes first.
-	return len(a) - len(b)
+	return len(a.lbls) - len(b.lbls)
+}
+
+// Copy labels from b on top of whatever was in ls previously, reusing memory or expanding if needed.
+func (ls *Labels) CopyFrom(b Labels) {
+	ls.lbls = append(ls.lbls[:0], b.lbls...)
+}
+
+// IsEmpty returns true if ls represents an empty set of labels.
+func (ls Labels) IsEmpty() bool {
+	return len(ls.lbls) == 0
+}
+
+// Call f on each label.
+func (ls Labels) Range(f func(l Label)) {
+	for _, l := range ls.lbls {
+		f(l)
+	}
+}
+
+// Call f on each label; if f returns non-nil then return that error.
+func (ls Labels) RangeToError(f func(l Label) error) error {
+	for _, l := range ls.lbls {
+		err := f(l)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// InternStrings calls intern on every string value inside ls, replacing them with what it returns.
+func (ls *Labels) InternStrings(intern func(string) string) {
+	for i, l := range ls.lbls {
+		ls.lbls[i].Name = intern(l.Name)
+		ls.lbls[i].Value = intern(l.Value)
+	}
+}
+
+// InternStrings calls release on every string value inside ls.
```

Comment:
```suggestion
// Release calls release on every string value inside ls.
```

I don't get if we need to really bind "release" semantics into this method. Perhaps `RangeStrings(func(string) string)` would be enough (for both intern and release methods)

Reply:
This is really temporary, since `InternStrings` is not a method we want with a more advanced change to Labels data structure. I guess I can comment that, but I don't want to put a lot of work into it.